bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVBridgeHandler.java

```diff
@@ -106,62 +105,50 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     private void initiateConnexion() {
         // Check if the provided api key is valid for use with the OpenUV service
-        getUVData("0", "0", null);
+        getUVData("0", "0", "0");
     }
 
-    public Map<ThingUID, @Nullable ServiceRegistration<?>> getDiscoveryServiceRegs() {
-        return discoveryServiceRegs;
-    }
-
-    public void setDiscoveryServiceRegs(Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs) {
-        this.discoveryServiceRegs = discoveryServiceRegs;
-    }
-
-    @Override
-    public void handleRemoval() {
-        // removes the old registration service associated to the bridge, if existing
-        ServiceRegistration<?> dis = getDiscoveryServiceRegs().get(getThing().getUID());
-        if (dis != null) {
-            dis.unregister();
-        }
-        super.handleRemoval();
-    }
-
-    public @Nullable OpenUVResult getUVData(String latitude, String longitude, @Nullable String altitude) {
-        StringBuilder urlBuilder = new StringBuilder(BASE_URL).append("?lat=").append(latitude).append("&lng=")
-                .append(longitude);
-
-        if (altitude != null) {
-            urlBuilder.append("&alt=").append(altitude);
-        }
-        String errorMessage = null;
+    public @Nullable OpenUVResult getUVData(String latitude, String longitude, String altitude) {
         try {
-            String jsonData = HttpUtil.executeUrl("GET", urlBuilder.toString(), header, null, null, REQUEST_TIMEOUT);
+            String jsonData = HttpUtil.executeUrl("GET", String.format(QUERY_URL, latitude, longitude, altitude),
+                    header, null, null, REQUEST_TIMEOUT_MS);
             OpenUVResponse uvResponse = gson.fromJson(jsonData, OpenUVResponse.class);
             if (uvResponse.getError() == null) {
                 updateStatus(ThingStatus.ONLINE);
                 return uvResponse.getResult();
             } else {
-                errorMessage = uvResponse.getError();
+                throw new OpenUVException(uvResponse.getError());
             }
         } catch (IOException e) {
-            errorMessage = e.getMessage();
+            logger.error("Error occured during API query : {}", e.getMessage());
```

Comment:
You are already setting the thing offline with the error message so you don't need to log it as well.
```suggestion
```